rfd/0010-api.md

```diff
@@ -123,21 +116,30 @@ func ValidateRole(r types.Role) error {
+  * `google.golang.org/grpc`
+  * `golang.org/x/net`
+  * `golang.org/x/crypto`
+  * `github.com/stretchr/testify`
 
-The new API client should be embedded in the `lib/auth.Client` during the deprecation process. That way new and existing endpoints can be added to the `api/client` and embedded in the `lib/auth.Client`.
+If the list of dependencies grows beyond 20 dependencies and indirect dependencies, this issue should be revisited for better tracking mechanisms.
 
-There is [issue](https://github.com/gravitational/teleport/issues/6394) used to track remaining endpoints that need to be converted to gRPC. 
+##### Deprecate the `lib/auth.Client`
+
+As stated before, all gRPC client related code will be moved out of `teleport/lib/auth/clt.go` and into `api/client`. Backwards compatibility will be maintained by embedding the gRPC `api/client.Client` in the internal `teleport/lib/auth.Client`.
 
-As this work is carried out, the `lib/auth.Client` should be adapted to attempt to make gRPC requests before falling back to HTTP in the case where the server version is older than the client version. These fallback methods should be moved to `lib/auth/httpfallback.go` to simplify the deprecation process.
+Additionally, the remaining HTTP endpoints in the internal client will be converted and added to the gRPC client. To maintain backwards compatibility with old servers, the internal client will be updated to first try the new gRPC endpoint before falling back to the existing HTTP endpoint. These fallback methods will be moved to `teleport/lib/auth/httpfallback.go` and deprecated after one major release cycle.
+
+There is [issue](https://github.com/gravitational/teleport/issues/6394) used to track remaining endpoints that need to be converted to gRPC. 
 
-Once all necessary endpoints have been added in gRPC, the gRPC client can completely replace the existing `lib/auth.Client`. At this point the underlying gRPC server will be the same, so the transition should be simple.
+Once all necessary endpoints have been added in gRPC and all HTTP fallback logic has been deprecated and removed, the gRPC client can completely replace the existing `lib/auth.Client`. At this point the underlying gRPC server will be the same, so the transition should be relatviley straightforward.
 
 ### api module
 
-The api package will be made a sub module and dependency of teleport.
+The api package will be made a sub module of teleport. It will exist in the `teleport/api`, and will be imported by the main `teleport` module through a go.mod [replace directive](https://golang.org/ref/mod#go-mod-file-replace) using a relative path.
 
-Users will be able to get the latest api version by using `go get github.com/teleport/api` or a specific version with `go get github.com/teleport/api@vX.Y.Z`..
+Users will be able to get the latest api version by using `go get github.com/gravitational/teleport/api`, or a specific version with `go get github.com/gravitational/teleport/api@vX.Y.Z`.
```

Comment:
Yes, mainly because part of our versioning promise is to maintain compatibility between "binaries", which includes the API. It's a lot easier to reason and maintain if it follows the same exact versioning. The main issue with this approach is that we may have extraneous releases with no actual API changes, in which case we could consider skipping some minor/patch releases for the API. I think major/minor versions should always line up though.

Reply:
we guarantee compatibility for binaries and gRPC API but we won't guarantee compatibility of the Go API under `api/`.
so while the gRPC API should remain compatible between v6.1 and v6.2, we can make it acceptable to break external Go code that depends on `teleport/api` when upgrading from v6.1 to v6.2 of the module.